kube-controllers/pkg/controllers/node/hostendpoints.go

```diff
@@ -17,300 +17,425 @@ package node
-				logrus.Infof("did not delete hostendpoint %q because it doesn't exist", hepName)
-				return nil
-			default:
-				logrus.WithError(err).Infof("failed to delete host endpoint %q, retrying", hepName)
-				time.Sleep(retrySleepTime)
-				continue
-			}
-		}
-		return nil
-	}
-	return fmt.Errorf("too many retries when deleting hostendpoint %q", hepName)
-}
-
-// isAutoHostendpoint determines if the given hostendpoint is managed by
+// isAutoHostEndpoint determines if the given HostEndpoint is managed by
 // kube-controllers.
-func isAutoHostendpoint(h *api.HostEndpoint) bool {
+func isAutoHostEndpoint(h *api.HostEndpoint) bool {
 	v, ok := h.Labels[hepCreatedLabelKey]
 	return ok && v == hepCreatedLabelValue
 }
 
-// createAutoHostendpoint creates an auto hostendpoint for the specified node.
-func (c *autoHostEndpointController) createAutoHostendpoint(ctx context.Context, n *libapi.Node) (*api.HostEndpoint, error) {
-	hep := c.generateAutoHostendpointFromNode(n)
-	rlKey := rateLimiterItemKey{Type: RateLimitCalicoCreate, Name: hep.Name}
-
-	time.Sleep(c.rl.When(rlKey))
-	res, err := c.client.HostEndpoints().Create(ctx, hep, options.SetOptions{})
+// createAutoHostEndpoint creates the supplied HostEndpoint
+func (c *autoHostEndpointController) createAutoHostEndpoint(hostEndpoint *api.HostEndpoint) error {
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+	_, err := c.client.HostEndpoints().Create(ctx, hostEndpoint, options.SetOptions{})
 	if err != nil {
 		logrus.Warnf("could not create hostendpoint for node: %v", err)
-		return nil, err
+		return err
 	}
-	c.rl.Forget(rlKey)
-	return res, nil
+	return nil
+}
+
+// generateDefaultAutoHostEndpointName returns the auto HostEndpoint name for default HostEndpoint
+func (c *autoHostEndpointController) generateDefaultAutoHostEndpointName(nodeName string) string {
+	return fmt.Sprintf("%s-%s", nodeName, hostEndpointNameSuffix)
 }
 
-// generateAutoHostendpointName returns the auto hostendpoint's name.
-func (c *autoHostEndpointController) generateAutoHostendpointName(nodeName string) string {
-	return fmt.Sprintf("%s-auto-hep", nodeName)
+// generateTemplateAutoHostEndpointName returns the name for HostEndpoint created with a template
+func (c *autoHostEndpointController) generateTemplateAutoHostEndpointName(nodeName string, templateName string) string {
+	return fmt.Sprintf("%s-%s-%s", nodeName, templateName, hostEndpointNameSuffix)
 }
 
-// getAutoHostendpointExpectedIPs returns all of the known IPs on the node resource
-// that should set on the auto hostendpoint.
-func (c *autoHostEndpointController) getAutoHostendpointExpectedIPs(node *libapi.Node) []string {
+// getAutoHostEndpointExpectedIPsMatchingTemplate finds the matching node IPs to the CIDRs specified in the template
+func (c *autoHostEndpointController) getAutoHostEndpointExpectedIPsMatchingTemplate(node *libapi.Node, template config.AutoHostEndpointTemplate) []string {
```

Comment:
```suggestion
func (c *autoHostEndpointController) getExpectedIPsMatchingTemplate(node *libapi.Node, template config.AutoHostEndpointTemplate) []string {
```

nit, but autoHostEndpoint is a bit redundant given the name of the controller - removing extra works helps make the code a smidge more readable.